src/Components/Web.JS/src/Boot.Web.ts

```diff
@@ -15,50 +15,113 @@ import { shouldAutoStart } from './BootCommon';
 import { Blazor } from './GlobalExports';
 import { WebStartOptions } from './Platform/WebStartOptions';
 import { attachStreamingRenderingListener } from './Rendering/StreamingRendering';
-import { attachProgressivelyEnhancedNavigationListener, detachProgressivelyEnhancedNavigationListener } from './Services/NavigationEnhancement';
+import { NavigationEnhancementCallbacks, attachProgressivelyEnhancedNavigationListener } from './Services/NavigationEnhancement';
 import { WebAssemblyComponentDescriptor } from './Services/ComponentDescriptorDiscovery';
-import { ServerComponentDescriptor, discoverComponents } from './Services/ComponentDescriptorDiscovery';
+import { ServerComponentDescriptor } from './Services/ComponentDescriptorDiscovery';
+import { DescriptorHandler, attachComponentDescriptorHandler, processComponentDescriptors } from './Rendering/DomMerging/BoundarySync';
+import { RootComponentManager } from './Services/RootComponentManager';
+import { RendererId } from './Rendering/RendererId';
 
 let started = false;
+let isPerformingEnhnacedNavigation = false;
```

Comment:
```suggestion
let isPerformingEnhancedNavigation = false;
```

Reply:
I'm amazed that you spotted this. Even with the change highlighted it took me 10 seconds to figure out what was wrong.